datafusion/core/src/datasource/physical_plan/parquet/mod.rs

```diff
@@ -93,6 +95,8 @@ pub struct ParquetExec {
     cache: PlanProperties,
     /// Options for reading Parquet files
     table_parquet_options: TableParquetOptions,
+    /// Optional user defined schema adapter
+    schema_adapter_factory: Option<Arc<dyn SchemaAdapterFactory>>,
```

Comment:
I think it is about time (not this PR) to create a `ParquetExecBuilder` so that new optional arguments can be aded without having to change all callsites.